app/lib/foreman_deployments/tasks/host_creation_task_definition.rb

```diff
@@ -30,6 +30,8 @@ def self.create_object(parameters)
 
       def self.initialize_manged_host(object, object_params)
         object_params['build'] ||= true
+        # Set attributes from a hostgroup
+        object.attributes = object.apply_inherited_attributes(object_params)
```

Comment:
Why not send the `object_params` from the get go? If the new statement will get all the parameters that need setting, it will do the inheritance logic by itself.